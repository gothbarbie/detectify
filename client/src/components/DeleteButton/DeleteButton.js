import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'

import Icon from '../Icon'

const Delete = styled.button`
  color: #ba3333;
  font-size: 1.3rem;
  font-weight: 700;
  border: none;
  background-color: transparent;
  cursor: pointer;
  display: flex;
  align-items: center;
  padding: 0;

  span {
    margin-right: 0.5rem;
  }
`

const DeleteButton = ({ onClick }) => (
  <Delete onClick={onClick}>
    <span>Delete Note</span> <Icon icon={['fas', 'times']} />
  </Delete>
)

DeleteButton.propTypes = {
  onClick: PropTypes.func.isRequired,
}

export default DeleteButton
