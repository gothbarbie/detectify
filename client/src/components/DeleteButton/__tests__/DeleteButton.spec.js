import React from 'react'
import { shallow } from 'enzyme'

import DeleteButton from '../DeleteButton'
describe('', () => {
  it('renders', () => {
    const props = {
      onClick: jest.fn(),
    }
    const wrapper = shallow(<DeleteButton {...props} />)

    expect(wrapper).toMatchSnapshot()
  })
})
