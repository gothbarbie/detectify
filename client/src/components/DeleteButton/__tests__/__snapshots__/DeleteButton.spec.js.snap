// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<styled.button
  onClick={[MockFunction]}
>
  <span>
    Delete Note
  </span>
   
  <Icon
    icon={
      Array [
        "fas",
        "times",
      ]
    }
  />
</styled.button>
`;
